lib/beaker/host/mac/pkg.rb

```diff
@@ -25,7 +27,8 @@ def generic_install_dmg(dmg_file, pkg_base, pkg_name)
     end
     dmg_name = File.basename(dmg_file, '.dmg')
     execute("hdiutil attach #{dmg_name}.dmg")
-    execute("installer -pkg /Volumes/#{pkg_base}/#{pkg_name} -target /")
+    # install the mounted *.pkg even if its basename is different from the *.dmg
+    execute("ls /Volumes/#{pkg_base}/*.pkg | xargs -t -I{} installer -pkg {} -target /")
```

Comment:
It's generally considered a bad practice to use `ls` in scripts. Since you're using a shell anyway, can't you use a for loop?

Reply:
Hi @ekohl.  Actually, just making a few more amendments to this..so that it will work if the pkg_name is valid.  Only if the pkg_name is not there, then search and install the *.pkg.  I'm just testing the change now.  

With regard to the `ls`; can you give me an example of how you mean with a for-loop?  I just want to find the *.pkg file that lives beneath /Volumes/#{pkg_base} and then install that.

Reply:
In other words, something like this:

```ruby
    execute("test -f #{specific_pkg_path}", :accept_all_exit_codes => true) do |result|
      if result.exit_code == 0
        # $pkg_name package found so install it
        execute("installer -pkg #{specific_pkg_path} -target /")
      else
        # else find and install any .pkg file in the volume
        execute("ls /Volumes/#{pkg_base}/*.pkg | xargs -t -I{} installer -pkg {} -target /")
      end
    end
```

Reply:
Changed to for-loop